shell.nix

```diff
@@ -0,0 +1,17 @@
+{ system ? builtins.currentSystem }:
```

Comment:
Is this file supposed to only use builtins?

Reply:
Yes, it shall have no other dependency.

I'd rather avoid public interfaces, where we can. Everything needed will be conveniently contained within `devshell` and `nixpkgs`.